### PR TITLE
enhancements to obrms to support optimization of pose alignment

### PR DESCRIPTION
**Pull request by dkoes**
This provides significantly more functionality to obrms, including the ability to calculate the minimum rms and different options for handling files

### CODE REVIEW

File: tools/obrms.cpp [hunk 1]
```diff
@@ -214,46 +291,122 @@ int main(int argc, char **argv)
...
+			exit(-1);
+		}
+	}
+	
 	//read reference
-	ifstream ifsref;
 	OBMol molref;
+	std::ifstream uncompressed_inmol(fileRef.c_str());
+	iostreams::filtering_stream<iostreams::input> ifsref;
+	string::size_type pos = fileRef.rfind(".gz");
```
Comment (ghutchis) [inline tools/obrms.cpp]: We have our own internal gzip-handling - do you need this?

File: tools/obrms.cpp [hunk 2]
```diff
@@ -168,32 +221,56 @@ static void processMol(OBMol& mol)
...
-	if (argc != 3 && argc != 4)
+	bool minimize = false;
+	bool separate = false;
+	bool help = false;
+	bool docross = false;
+	string fileRef;
+	string fileTest;
+	string fileOut;
+	
+	program_options::options_description desc("Allowed options");
```
Comment (ghutchis) [inline tools/obrms.cpp]: I'm not going to add Boost to this for the sake of adding help options.

**Comment by ghutchis** [comment]
I like the concept, but I'd really like to avoid adding Boost just for this.

**Comment by dkoes** [comment]
Drat, I was hoping to sneak it in.  In my copious free time I will see if I can refactor the code to avoid boost, but the cmake files should be configured so obrms just isn't built if boost isn't available.

**Comment by dkoes** [comment]
In what proved to be incredibly poor time management on my part, I have removed the boost dependency.

**Comment by ghutchis** [comment]
I can't merge this with the Windows compile fail:
```
C:\openbabel\tools\obrms.cpp(90): error C2057: expected constant expression [C:\openbabel\build\tools\obrms.vcxproj]
613C:\openbabel\tools\obrms.cpp(90): error C2466: cannot allocate an array of constant size 0 [C:\openbabel\build\tools\obrms.vcxproj]
614C:\openbabel\tools\obrms.cpp(90): error C2133: 'refcoord' : unknown size [C:\openbabel\build\tools\obrms.vcxproj]
615C:\openbabel\tools\obrms.cpp(91): error C2057: expected constant expression [C:\openbabel\build\tools\obrms.vcxproj]
616C:\openbabel\tools\obrms.cpp(91): error C2466: cannot allocate an array of constant size 0 [C:\openbabel\build\tools\obrms.vcxproj]
617C:\openbabel\tools\obrms.cpp(91): error C2133: 'testcoord' : unknown size [C:\openbabel\build\tools\obrms.vcxproj]
```

**Comment by dkoes** [comment]
Do you know if windows supports alloca?

**Comment by dkoes** [comment]
Any idea why these builds are failing? calc_rms and rotate_coords are defined in obutil.  I don't see an obvious guard that would change this on other platforms/compilers.

**Comment by baoilleach** [comment]
I'll have a look now.

**Comment by baoilleach** [comment]
Looks like it's somehow related to older compilers. Let me try with an older one...

**Comment by baoilleach** [comment]
Builds for me with MSVC2017, 2015 and 2013. Appveyor is running with MSVC2012, and giving the same failure as the Linux build. I'll try an old gcc compiler so...

**Comment by baoilleach** [comment]
The penny drops...it just needed to be rebased on master due to changes in the interim. I've sent a PR that adds the necessary include which hopefully should fix everything.

**Comment by baoilleach** [comment]
By the way, any chance of some Sphinx docs on the topic? Right now, the only documented executable is babel/obabel. I've often wondered what those other executables do.

**Comment by dkoes** [comment]
I've submitted a pull request with an updated man page, which probably isn't what you are asking for but isn't nothing.

**Comment by baoilleach** [comment]
True and true :-)

**Pull request merged**